### ISSUE TITLE
Outputting XML and writing XML?

### ISSUE THREAD

**Issue by skipallmighty** [drmohundro/SWXMLHash#20]
Beautifully written library. Congratulations!
I noticed that I can assign in new value to the text of an element, because it's a `var` string optional, so that works great. I hope I'm doing that the way that you intended.

```swift
let filePath = NSBundle(forClass: self.classForCoder).pathForResource("cds", ofType: "xml")
var contents = File.open(filePath!, utf8: NSUTF8StringEncoding)!
let xml = SWXMLHash.parse(contents)
println(xml["CATALOG"]["CD"][0]["TITLE"].element?.text!)
// is this how I should set the new text?
xml["CATALOG"]["CD"][0]["TITLE"].element?.text = "New Text"
//Output XML here somehow. 
```

I'm wondering if I can output the new XML so I can save it to document? 

I understand the objects that `all` and `children` return.  I was about to write something that would output the new XML with the new values. I wasn't sure if there was a easy way to get that without having to write something special that outputs XML.

For this example I am using this XML as an example: 
http://www.w3schools.com/xml/cd_catalog.xml

**Comment by drmohundro** [comment]
Good question! At the moment, the library is read-only - no XML writing support currently exists. The `text` property is a var primarily because of how I'm currently parsing the XML (using `NSXMLParser`). I create the XMLElement at the time the tag is parsed, which is before I know if any text is set for the element.

I'd have to put some thought into what a writeable API would look like, though. Ideally, such an API would allow for building XML without first starting with a document (i.e. no call to `SWXMLHash.parse` at the beginning).

I'll probably hold off on this unless a lot of other people request it or if someone wanted open a pull request for it.

Hope this helps to answer your question!

**Comment by skipallmighty** [comment]
If I put together a pull request would you be interested? I believe what you need is a sort of `toString()` or adopt the `Printable` protocol to at least get the XML back as a first step. It seems you can already alter the data on a minimal level. Also you can append to those lists you return. It's already in pretty good shape but I just need that data back.

**Comment by drmohundro** [comment]
Sure, that'd be great - it'd certainly be a start and, even if it the writeable support isn't heavily used, it would certainly help with debugging. I'll take a look at your pull request, thanks!

**Comment by drmohundro** [comment]
I've merged #22 so I'll mark this one closed for now.

**Comment by skipallmighty** [comment]
Fantastic thanks! :+1:
